### ISSUE TITLE
[New Check] warn when using parameter expansion `-` instead of `:-` where the parameter is unconditionally set

### ISSUE THREAD

**Issue by dejj** [koalaman/shellcheck#2440]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

HEADERS=$(curl --head --silent "http://www.example.com")
HTTPSTATUS=$(echo "${HEADERS}" | grep HTTP | cut -d' ' -f2)
if [ "${HTTPSTATUS-999}" -lt 400 ]; then
  echo "ok"
else
  echo "HTTP error code received ${HTTPSTATUS}"
fi
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Expansion operator '-' can never succeed, because parameter HTTPSTATUS is set on all incoming control flows. Did you mean ':-' ?

### Rationale
The author of that code seemed to want to substitute a missing value for `HTTPSTATUS` with `999`, so the test would fail.
Bash defines that `:-` substitutes, if the parameter is unset or null.
It also defines that `-` substitutes, if the parameter is unset, but does not substitute null.
The explanation is only in text form in the 6th paragraph of: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html ("...if the colon is omitted, the operator tests only for existence.")
Some readers may also confuse `-` with arithmetic subtraction.
